c7n/resources/ec2.py

```diff
@@ -197,7 +197,13 @@ def get_resource_date(self, i):
         v = i.get('StateTransitionReason')
         if not v:
             return None
-        return parse(self.RE_PARSE_AGE.findall(v)[0][1:-1])
+        dates = self.RE_PARSE_AGE.findall(v)
+        if dates:
+            try:
+                return parse(dates[0][1:-1])
+            except ValueError:
+                self.log.warning('Failed to parse date in EC2 state-age ({})'.format(v))
```

Comment:
ok I've changed it so it does not catch that exception.  My thinking was that you were trying to error proof this method.  

Just so I understand your approach for future code, how does it differ if the RE fails to match (we now `return None`, effective as of this PR) vs. the date being unable to be parsed (we reraise the exception).

The original purpose of this PR was to not raise an exception when the date came in as "Server.InternalError", and that is done.  But now if we get an error message with parentheses in it we'll raise an exception, e.g:

```
>>> error = "Server.InternalError (Error code 42)"
>>> StateTransitionAge(None).get_resource_date({'StateTransitionReason': error})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "c7n/resources/ec2.py", line 202, in get_resource_date
    return parse(dates[0][1:-1])
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/lib/python2.7/site-packages/dateutil/parser.py", line 1168, in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/lib/python2.7/site-packages/dateutil/parser.py", line 559, in parse
    raise ValueError("Unknown string format")
ValueError: Unknown string format

```